### ISSUE TITLE
SKStoreReviewController is deprecated in iOS 18, and replacement is not accesible in MAUI.

### ISSUE THREAD

**Issue by PureWeen** [xamarin/xamarin-macios#21410]
Moved from https://github.com/dotnet/maui/issues/25169

### Description

Hello,

with iOS 18 it seems that Apple deprecated SKStoreReviewController. Now every plugin and call made from SKStoreReviewController are not working. There is a replacement with AppStore.requestReview(in: scene). But from MAUI it is inaccessible. 

https://developer.apple.com/documentation/storekit/appstore/3954432-requestreview/

### Steps to Reproduce

Create an application
on ios18 invoke
```
if (UIApplication
                            .SharedApplication?
                            .ConnectedScenes?
                            .ToArray<UIScene>()?
                            .FirstOrDefault(x => x.ActivationState == UISceneActivationState.ForegroundActive) is UIWindowScene windowScene)
                    {
                        SKStoreReviewController.RequestReview(windowScene);
                        return Task.CompletedTask;
                    }
```
see that application hangs.

same code on iOS 17.5 works, create a review popup or just passes quietly.

### Version with bug

8.0.91 SR9.1

### Is this a regression from previous behavior?

No, this is something new

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 18+

### Did you find any workaround?

No

**Comment by rolfbjarne** [comment]
AppStore.RequestReview looks like a really simple API, so we might be able to bind it manually.

I'll see what I can do.

**Comment by rolfbjarne** [comment]
We're binding this as an experimental API named `StoreKit.AppStore.RequestReview(UIWindowScene)`.

**Comment by MerSpyro** [comment]
@rolfbjarne @PureWeen
Could you provide information when and where this fix will be available? Im not too sure where I'd go to look for this, in .net release, MAUI release or somewhere else?

**Comment by rolfbjarne** [comment]
> @rolfbjarne @PureWeen Could you provide information when and where this fix will be available? Im not too sure where I'd go to look for this, in .net release, MAUI release or somewhere else?

This will be in an update to the iOS workload, most likely when we add support for Xcode 16.1 (which will probably happen soon after .NET 9 has been released, although this depends a bit on when exactly Apple makes Xcode 16.1 stable).

**Comment by tipa** [comment]
Can the same be considered for the [reloadAllTimelines()](https://developer.apple.com/documentation/widgetkit/widgetcenter/reloadalltimelines()) method? 
I am currently using [this workaround](https://github.com/xamarin/xamarin-macios/issues/15315) (in combination with an app extension written in Swift) but it adds quite a bit of boilerplate to my projects.

**Comment by rolfbjarne** [comment]
> Can the same be considered for the [reloadAllTimelines()](https://developer.apple.com/documentation/widgetkit/widgetcenter/reloadalltimelines()) method? I am currently using [this workaround](https://github.com/xamarin/xamarin-macios/issues/15315) (in combination with an app extension written in Swift) but it adds quite a bit of boilerplate to my projects.

I filed a suggestion for this: https://github.com/xamarin/xamarin-macios/issues/21466

**Comment by tipa** [comment]
@rolfbjarne I tried the new API and it doesn't seem to work for me. It is [documented](https://developer.apple.com/documentation/storekit/appstore/3954432-requestreview#4278439) that the review dialog is shown always in development mode, but when I call the method, nothing happens at all: no dialog, no crash, no logged error (that I could see) - both on macOS and iOS.

I then tried binding the method myself - using the same approach that is explained [here](https://github.com/xamarin/xamarin-macios/issues/15315):

```
@MainActor
@_cdecl("RequestReview")
@available(iOS 16, macCatalyst 16, *)
public func StoreKit_RequestReview(scene: UIWindowScene)
{
    AppStore.requestReview(in: scene)
}
```

And in my C# code:

```
[DllImport("__Internal", EntryPoint = "RequestReview")]
public static extern int RequestReview(IntPtr windowScene);

...

var scene = UIApplication.SharedApplication.ConnectedScenes.ToArray()
    .FirstOrDefault(x => x.ActivationState == UISceneActivationState.ForegroundActive);
if (scene is UIWindowScene windowScene) { RequestReview(windowScene.Handle); }
```

And using that code, the rating dialog shows reliably.

**Comment by rolfbjarne** [comment]
@tipa that's strange, this code works for me:

```cs
var scene = UIApplication.SharedApplication.ConnectedScenes.ToArray()
				.FirstOrDefault (x => x.ActivationState == UISceneActivationState.ForegroundActive);
if (scene is UIWindowScene windowScene) {
	StoreKit.AppStore.RequestReview (windowScene);
}
```

Complete test project: [ios-plain-80b52c1.zip](https://github.com/user-attachments/files/17993871/ios-plain-80b52c1.zip)

**Comment by tipa** [comment]
@rolfbjarne I tested your project and it works. It now also works when I call the API from my iOS apps. I can't say yet, why this wasn't the case when I tested it last week.

However, I can still reproduce the problem on macOS. Can you have a look at my test project?
The review dialog only appears when I call the self-bound method.

[reviewmacos.zip](https://github.com/user-attachments/files/17994921/reviewmacos.zip)

**Comment by rolfbjarne** [comment]
OK, I see what's happening.

Adding this to the csproj should work around it:

```xml
<ItemGroup>
  <ReferenceNativeSymbol Include="XamarinSwiftFunctions" SymbolType="ObjectiveCClass" />
</ItemGroup>
```

**Comment by rolfbjarne** [comment]
> @rolfbjarne I tested your project and it works. It now also works when I call the API from my iOS apps. I can't say yet, why this wasn't the case when I tested it last week.
> 
> However, I can still reproduce the problem on macOS. Can you have a look at my test project? The review dialog only appears when I call the self-bound method.
> 
> [reviewmacos.zip](https://github.com/user-attachments/files/17994921/reviewmacos.zip)

I've merged a fix for this, which will be included when we ship support for Xcode 16.2.
